### DISCUSSION TITLE
Rendering sometimes not coming back

### DISCUSSION THREAD

**Discussion by Tillerz** [vladmandic/sdnext#668] [Q&A] [answered]
For a few days (2 or 3?) I have the problem that the rendering isn't coming back to the UI sometimes. I can render a dozen images in a row (single renders each time, I don't do batches etc) before it breaks, sometimes it breaks after 3. I had no such issues when I started with automatic after I got rid of the black images by fixing the settings.

In the console window it goes to 100%, but in the UI it sticks at 96%. I tried to reload the UI <ctrl-f5> and/or open another one in a fresh window, it sticks with the stop/skip buttons. I got it once to show "generate" again this way, but clicking that only said "added to queue", so actually the backend is stuck, not the UI. Only way for me to solve it is to kill the shell window and restart the server.

I always do an immediate upscale to 1024x1024 and have the aesthetic scorer enabled. Disabling the scorer didn't improve it, so that's not it.

It's also not the model and not the VAE, the same happens with others, too. It's hanging when the second bar in the console is at 100%, so at the end of the upscaling.

Is there a way to enable some logging/debug output or something so I can see at which part of the rendering it actually breaks?

my settings:
```
Steps: 40, Sampler: Euler a, CFG scale: 10, Face restoration: CodeFormer,
Size: 512x512, Model: anim-artius-realistic-hd-zovya, VAE: orangemix.vae,
Denoising strength: 0.55, Clip skip: 1, Hires resize: 1024x1024,
Hires upscaler: Latent
```

**Comment by vladmandic** [comment, marked as answer]
first thing make sure this is set: settings -> live preview -> approx nn.
second, there are open issues for this, there is a torch race condition that i'm trying to isolate.

**Comment by Tillerz** [comment]
The approx nn already fix it, thanks a lot. :)
